rdkit/Chem/Draw/IPythonConsoleIntegration.py

```diff
@@ -0,0 +1,602 @@
+    
+    uid = molViewState.uid
+    
+    # Required global objects
+    globals()['rdkit_3dviewer_start_button_clicked_'+uid] = False
+    globals()['drawAs_no_wg_'+uid] = drawAs
+    globals()['pStyle_no_wg_'+uid] = pStyle
+    
+    keys=molViewState.moldict.keys()
+    
+    # Right hand panel (widgets)
+    
+    wgListBox=list()
+    
+    itemLayout=Layout(display='flex', flex_flow='row', justify_content='space-between')
+    
+    wgListBox.append(Box([Label(value='uid'), HTML(description='', value=str(uid))], layout=itemLayout))
+    
+    # Add protein
+    if protein is not None:
+        #print 'adding protein'
+        globals()['protein_'+uid] = protein
+    
+    
+    # Users are expected to have keys int. 
+    # If keys are not str then still the rendering is possible but ChangeActiveLigand() wont't work currently
+    if type(keys[0])=='str': 
+        mol_id_ini = keys[0]
+    else:
+        mol_id_ini = str(keys[0])
+        
+    globals()['selected_mols_view_'+uid] = HTML(description='', value=mol_id_ini)
+    globals()['selected_confs_view_'+uid] = HTML(description='', value='0')
+    globals()['molId_'+uid] = Dropdown(description='', options=keys,value=keys[0])
+    globals()['selectMultiMols_'+uid] = Checkbox(description='selectMultiMols', value=False)
+    globals()['selectAllMols_'+uid] = Checkbox(description='selectAllMols', value=False)
+    globals()['confId_'+uid] = Dropdown(description='', options=list(range(9)), value=0)
+    globals()['selectMultiConfs_'+uid] = Checkbox(description='selectMultiConfs', value=False)
+    globals()['selectAllConfs_'+uid] = Checkbox(description='selectAllConfs', value=False)
+    
+    
+    wgListBox.append(Box([Label(value='Mols'),globals()['selected_mols_view_'+uid]], layout=itemLayout))
+    wgListBox.append(Box([Label(value='Confs'),globals()['selected_confs_view_'+uid]], layout=itemLayout))
+    
+    wgListBox.append(Box([Label(value='molId'),globals()['molId_'+uid]], layout=itemLayout))
+    
+    cbMolSelect=Box([Label(value=''),
+                     HBox([globals()['selectMultiMols_'+uid], 
+                           globals()['selectAllMols_'+uid]])
+                    ], layout=itemLayout)
+    
+    wgListBox.append(cbMolSelect)
+    
+    wgListBox.append(Box([Label(value='confId'),globals()['confId_'+uid]], layout=itemLayout))
+    
+    cbConfSelect=Box([Label(value=''),
+                      HBox([globals()['selectMultiConfs_'+uid], 
+                            globals()['selectAllConfs_'+uid]])
+                     ], layout=itemLayout)
+    
+    wgListBox.append(cbConfSelect)
+    
+    
+    
+    rdkitWG[globals()['molId_'+uid]._model_id] = uid
+    rdkitWG[globals()['selectMultiMols_'+uid]._model_id] = uid
+    rdkitWG[globals()['selectAllMols_'+uid]._model_id] = uid
+    rdkitWG[globals()['confId_'+uid]._model_id] = uid
+    rdkitWG[globals()['selectMultiConfs_'+uid]._model_id] = uid
+    rdkitWG[globals()['selectAllConfs_'+uid]._model_id] = uid
```

Comment:
How important is it to pass the `uid` to all these controls? Is it maybe already available in all parts of your code where this is used?